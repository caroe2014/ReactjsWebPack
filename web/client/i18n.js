// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import i18next from 'i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import Backend from 'i18next-locize-backend';

/* istanbul ignore next */
i18next
  .use(LanguageDetector)
  .use(Backend)
  .init({
    // fallback depending on user language
    fallbackLng: {
      'ja': ['ja-JP'],
      'default': ['en-US']
    },
    order: ['querystring', 'navigator'],
    lookupQuerystring: 'lng',
    allowMultiLoading: true,
    backend: {
      projectId: '838d5fce-27b5-4368-8c54-8fcb33577f9a',
      referenceLng: 'en-US',
      version: process.env.ENVIRONMENT === 'production' ? 'latest' : process.env.ENVIRONMENT || process.env.NODE_ENV
    },
    debug: false,
    detection: {
      caches: []
    },
    // have a common namespace used around the full app
    ns: ['core', `domain-${window.location.hostname}`],
    defaultNS: `domain-${window.location.hostname}`,
    fallbackNS: [`domain-${window.location.hostname}`, 'core'],

    keySeparator: false, // we use content as keys

    interpolation: {
      formatSeparator: ','
    },

    react: {
      wait: true
    }
  }, (err, t) => {
    if (err) return console.log('Something went wrong while loading.', err);
  });

export default i18next;
