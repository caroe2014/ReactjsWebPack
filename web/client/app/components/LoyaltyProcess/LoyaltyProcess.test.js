import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { ThemeProvider } from 'styled-components';
import theme from 'web/client/theme.js';
import i18n from 'i18next';
import LoyaltyProcess from '.';
import jsdom from 'jsdom';

configure({ adapter: new Adapter() });

/* global describe, it, expect, beforeEach, jest */
/* eslint-disable max-len */
global.window = new jsdom.JSDOM().window;
global.document = window.document;
global.scrollTo = jest.fn();

jest.mock('web/client/i18n', () => ({
  t: (key) => {
    return key;
  }
}));
jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate HoC receive the t function as a prop
  withTranslation: () => Component => {
    Component.defaultProps = { ...Component.defaultProps, t: (key) => key };
    return Component;
  }
}));

jest.mock('web/client/app/reduxpages/ConnectedComponents', () => {
  return {
    ConnectedLoyaltyProcess: <div/>
  };
});

jest.mock('react-i18next', () => ({
  Trans: 'div'
}));

describe('LoyaltyProcess', () => {
  let mockFn = jest.fn();
  let wrapper;
  let loyaltyProcess = {
    accrueError: true,
    loyaltyLinkedAccounts: [
      {
        accountNumber: '111'
      }
    ]
  };
  let loyaltyDetails = {
    header: 'test'
  };
  let loyaltyInfoMap = {
    1001: {
      id: 'test',
      loyaltyInfo: {
        test: 'test'
      }
    }
  };
  let siteId = 1001;
  let cartLoyaltyInfo = {
    inquiryId: 1001
  };
  beforeEach(() => {
    wrapper = shallow(<LoyaltyProcess
      cartLoyaltyInfo={cartLoyaltyInfo}
      loyaltyProcess={loyaltyProcess}
      loyaltyDetails={loyaltyDetails}
      loyaltyInfoMap={loyaltyInfoMap}
      siteId={siteId}
      fetching={false}
      showAccount
      cancelLoyalty={false}/>);
  });

  it('does a snapshot test', () => {
    const snapwrapper = mount(<ThemeProvider theme={theme}>
      <LoyaltyProcess
        cartLoyaltyInfo={cartLoyaltyInfo}
        inquiryError={'someError'}
        loyaltyProcess={loyaltyProcess}
        loyaltyDetails={loyaltyDetails}
        loyaltyInfoMap={loyaltyInfoMap}
        siteId={siteId}
        fetching={false}
        showAccount
        cancelLoyalty={false}/>
    </ThemeProvider>);
    expect(toJson(snapwrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('should render without throwing an error when cartLoyaltyInfo is unavailable and inquiryError is present', () => {
    const newWrapper = shallow(<LoyaltyProcess
      cartLoyaltyInfo
      inquiryError={'someError'}
      disableLoyalty={mockFn}
      loyaltyProcess={loyaltyProcess}
      loyaltyDetails={loyaltyDetails}
      loyaltyInfoMap={loyaltyInfoMap}
      siteId={siteId}
      fetching={false}
      showAccount
      cancelLoyalty={false}/>);
    expect(newWrapper.exists()).toBe(true);
  });

  it('should call onRetry when retry is clicked on inquiryError', () => {
    const newWrapper = shallow(<LoyaltyProcess
      cartLoyaltyInfo
      inquiryError={'someError'}
      sendLoyaltyInquiry={mockFn}
      sendLoyaltyInfo={mockFn}
      disableLoyalty={mockFn}
      loyaltyProcess={loyaltyProcess}
      loyaltyDetails={loyaltyDetails}
      loyaltyInfoMap={loyaltyInfoMap}
      siteId={siteId}
      fetching={false}
      showAccount
      cancelLoyalty={false}/>);

    const spy = jest.spyOn(newWrapper.instance(), 'onReEnterClick');
    newWrapper.instance().forceUpdate();
    newWrapper.find('.re-enter-button').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });

  // it('should call onRetry when retry is clicked on error', () => {
  //   const newWrapper = shallow(<LoyaltyProcess
  //     cartLoyaltyInfo
  //     error={'someError'}
  //     sendLoyaltyInquiry={mockFn}
  //     sendLoyaltyInfo={mockFn}
  //     loyaltyProcess={loyaltyProcess}
  //     loyaltyDetails={loyaltyDetails}
  //     loyaltyInfoMap={loyaltyInfoMap}
  //     siteId={siteId}
  //     fetching={false}
  //     showAccount
  //     cancelLoyalty={false}/>);

  //   const spy = jest.spyOn(newWrapper.instance(), 'onRetry');
  //   newWrapper.instance().forceUpdate();
  //   newWrapper.find('.re-enter-button').simulate('click');
  //   expect(spy).toHaveBeenCalled();
  //   spy.mockClear();
  // });

  it('should render without throwing an error when cancelLoyalty is true', () => {
    loyaltyProcess.processAccrue = true;
    const newWrapper = shallow(<LoyaltyProcess
      cartLoyaltyInfo
      loyaltyProcess={loyaltyProcess}
      disableLoyalty={mockFn}
      loyaltyDetails={loyaltyDetails}
      loyaltyInfoMap={loyaltyInfoMap}
      siteId={siteId}
      fetching={false}
      showAccount
      cancelLoyalty/>);
    expect(newWrapper.exists()).toBe(true);
  });

  describe('LoyaltyProcess with props loyaltyProcess', () => {
    let newLoyaltyProcess = {
      showAccountSelection: true,
      loyaltyLinkedAccounts: []
    };
    it('should render without throwing an error', () => {
      const newWrapper = shallow(<LoyaltyProcess
        cartLoyaltyInfo={cartLoyaltyInfo}
        loyaltyProcess={newLoyaltyProcess}
        disableLoyalty={mockFn}
        loyaltyDetails={loyaltyDetails}
        loyaltyInfoMap={loyaltyInfoMap}
        siteId={siteId}
        fetching={false}
        showAccount
        cancelLoyalty={false}/>);
      expect(newWrapper.exists()).toBe(true);
    });

    it('should render without throwing an error when one loyaltyLinkedAccounts is available', () => {
      newLoyaltyProcess.loyaltyLinkedAccounts = [
        {
          accountNumber: '111'
        }
      ];
      const newWrapper = shallow(<LoyaltyProcess
        cartLoyaltyInfo={cartLoyaltyInfo}
        loyaltyProcess={newLoyaltyProcess}
        disableLoyalty={mockFn}
        loyaltyDetails={loyaltyDetails}
        loyaltyInfoMap={loyaltyInfoMap}
        siteId={siteId}
        fetching={false}
        showAccount
        cancelLoyalty={false}/>);
      expect(newWrapper.exists()).toBe(true);
    });

    it('should render without throwing an error when multiple loyaltyLinkedAccounts are available', () => {
      newLoyaltyProcess.loyaltyLinkedAccounts = [
        {
          accountNumber: '111'
        },
        {
          accountNumber: '112'
        }
      ];
      const newWrapper = shallow(<LoyaltyProcess
        cartLoyaltyInfo={cartLoyaltyInfo}
        loyaltyProcess={newLoyaltyProcess}
        loyaltyDetails={loyaltyDetails}
        disableLoyalty={mockFn}
        loyaltyInfoMap={loyaltyInfoMap}
        siteId={siteId}
        fetching={false}
        showAccount
        cancelLoyalty={false}/>);
      expect(newWrapper.exists()).toBe(true);
    });

    it('should render without throwing an error when Points are already Accrued', () => {
      let newLoyaltyProcess = {
        showAccountSelection: false,
        loyaltyLinkedAccounts: [
          {
            accountNumber: '111'
          },
          {
            accountNumber: '112'
          }
        ],
        isPointAccrued: true
      };
      const newWrapper = shallow(<LoyaltyProcess
        cartLoyaltyInfo={cartLoyaltyInfo}
        loyaltyProcess={newLoyaltyProcess}
        loyaltyDetails={loyaltyDetails}
        disableLoyalty={mockFn}
        loyaltyInfoMap={loyaltyInfoMap}
        siteId={siteId}
        fetching={false}
        showAccount
        cancelLoyalty={false}/>);
      expect(newWrapper.exists()).toBe(true);
    });

    it('should render without throwing an error when cartloyaltyinfo is not present', () => {
      let newLoyaltyProcess = {
        showAccountSelection: false,
        loyaltyLinkedAccounts: [
          {
            accountNumber: '111'
          },
          {
            accountNumber: '112'
          }
        ],
        isPointAccrued: true
      };
      const newWrapper = shallow(<LoyaltyProcess
        loyaltyProcess={newLoyaltyProcess}
        loyaltyDetails={loyaltyDetails}
        loyaltyInfoMap={loyaltyInfoMap}
        disableLoyalty={mockFn}
        siteId={siteId}
        fetching={false}
        showAccount
        cancelLoyalty={false}/>);
      expect(newWrapper.exists()).toBe(true);
    });

  });
});
