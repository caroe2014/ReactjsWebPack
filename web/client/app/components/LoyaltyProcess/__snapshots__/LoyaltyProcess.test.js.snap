// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoyaltyProcess does a snapshot test 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "addOnTab": "#347CFF",
        "addToCart": "#347CFF",
        "addonBg": "#EEEEEE",
        "backgroundGrey": "#E6E6E6",
        "black": "#000000",
        "close": "red",
        "contextBackground": "#f0f0f3",
        "darkGrey": "#464646",
        "disableBackground": "#E4E4E4",
        "disableButton": "#D1D1D1",
        "error": "#cc0000",
        "footer": "#48a942",
        "glutenFree": "#8d7356",
        "healthy": "#359919",
        "hint": "#a6a6a6",
        "inputBottomBorder": "lightgray",
        "itemBorder": "lightgrey",
        "itemTile": "#3C474C0",
        "light": "white",
        "loader": "#ababab",
        "loginBackground": "#303030",
        "loginBoxColor": "#434149",
        "loginField": "#AAA7AC",
        "loginFormBackground": "#484848",
        "navigationMenu": "#48a942",
        "navigationMenuHover": "cornflowerblue",
        "new": "#ff6000",
        "open": "#48a942",
        "paymentSuccessTitle": "#505050",
        "placeHolder": "#909090",
        "primary": "#347CFF",
        "primaryButton": "#347CFF",
        "primaryText": "#383838",
        "secondaryText": "#a6a6a6",
        "spicy": "#ed1c24",
        "text": "black",
        "textGrey": "#505050",
        "tile": "#505050",
        "validationError": "#cc0000",
        "vegan": "#0d4a22",
        "vegetarian": "#0d4a22",
      },
      "fontFamily": "Roboto",
      "fontSize": Object {
        "lg": "1.8rem",
        "md": "1.3rem",
        "nm": "1rem",
        "sm": "0.75rem",
        "xl": "2.4rem",
      },
      "fontSizeFromProps": [Function],
      "mediaBreakpoints": Object {
        "desktop": [Function],
        "mobile": [Function],
        "tablet": [Function],
      },
      "nav": Object {
        "height": "70px",
      },
      "receiptFontFamily": "Roboto",
      "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
      "spacing": Object {
        "lg": "4em",
        "md": "2em",
        "nm": "1em",
        "none": "0em",
        "sm": "0.5em",
        "xl": "8em",
      },
    }
  }
>
  <LoyaltyProcess
    cancelLoyalty={false}
    cartLoyaltyInfo={
      Object {
        "inquiryId": 1001,
      }
    }
    fetching={false}
    inquiryError="someError"
    loyaltyDetails={
      Object {
        "header": "test",
      }
    }
    loyaltyInfoMap={
      Object {
        "1001": Object {
          "id": "test",
          "loyaltyInfo": Object {
            "test": "test",
          },
        },
      }
    }
    loyaltyProcess={
      Object {
        "accrueError": true,
        "loyaltyLinkedAccounts": Array [
          Object {
            "accountNumber": "111",
          },
        ],
      }
    }
    showAccount={true}
    siteId={1001}
  >
    <Flex
      className="container-parent"
      id="accrual-modal"
    >
      <Box
        className="container-parent sc-bwzfXH hKiLMS"
        id="accrual-modal"
      >
        <div
          className="container-parent sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
          id="accrual-modal"
        >
          <Styled(Fixed)
            className="guest-profile-modal-background"
          >
            <Fixed
              0="s"
              1="p"
              2="a"
              3="c"
              4="e"
              className="guest-profile-modal-background sc-kfGgVZ erxrZb"
              extend={[Function]}
            >
              <ForwardRef
                0="s"
                1="p"
                2="a"
                3="c"
                4="e"
                className="guest-profile-modal-background sc-kfGgVZ erxrZb sc-ksYbfQ kYqsUP"
                extend={[Function]}
              >
                <Position
                  0="s"
                  1="p"
                  2="a"
                  3="c"
                  4="e"
                  className="guest-profile-modal-background sc-kfGgVZ erxrZb sc-ksYbfQ kYqsUP"
                >
                  <ForwardRef
                    0="s"
                    1="p"
                    2="a"
                    3="c"
                    4="e"
                    className="guest-profile-modal-background sc-kfGgVZ erxrZb sc-ksYbfQ kYqsUP sc-TOsTZ kjYrri"
                  >
                    <div
                      0="s"
                      1="p"
                      2="a"
                      3="c"
                      4="e"
                      className="guest-profile-modal-background sc-kfGgVZ erxrZb sc-ksYbfQ kYqsUP sc-TOsTZ kjYrri"
                    />
                  </ForwardRef>
                </Position>
              </ForwardRef>
            </Fixed>
          </Styled(Fixed)>
          <Styled(Modal)
            className="loyalty-process-container"
            onKeyDown={[Function]}
            role="dialog"
          >
            <Modal
              0="s"
              1="p"
              2="a"
              3="c"
              4="e"
              bg="white"
              borderRadius={2}
              className="loyalty-process-container sc-hwwEjo kitExE"
              extend={[Function]}
              left="50%"
              maxHeight="100vh"
              maxWidth="100vw"
              onKeyDown={[Function]}
              p={3}
              role="dialog"
              top="50%"
            >
              <ForwardRef
                0="s"
                1="p"
                2="a"
                3="c"
                4="e"
                bg="white"
                borderRadius={2}
                className="loyalty-process-container sc-hwwEjo kitExE sc-frDJqD fcTlLW"
                extend={[Function]}
                left="50%"
                maxHeight="100vh"
                maxWidth="100vw"
                onKeyDown={[Function]}
                p={3}
                role="dialog"
                top="50%"
              >
                <Fixed
                  0="s"
                  1="p"
                  2="a"
                  3="c"
                  4="e"
                  bg="white"
                  borderRadius={2}
                  className="loyalty-process-container sc-hwwEjo kitExE sc-frDJqD fcTlLW"
                  extend={[Function]}
                  left="50%"
                  maxHeight="100vh"
                  maxWidth="100vw"
                  onKeyDown={[Function]}
                  p={3}
                  role="dialog"
                  top="50%"
                >
                  <ForwardRef
                    0="s"
                    1="p"
                    2="a"
                    3="c"
                    4="e"
                    bg="white"
                    borderRadius={2}
                    className="loyalty-process-container sc-hwwEjo kitExE sc-frDJqD fcTlLW sc-ksYbfQ kYqsUP"
                    extend={[Function]}
                    left="50%"
                    maxHeight="100vh"
                    maxWidth="100vw"
                    onKeyDown={[Function]}
                    p={3}
                    role="dialog"
                    top="50%"
                  >
                    <Position
                      0="s"
                      1="p"
                      2="a"
                      3="c"
                      4="e"
                      bg="white"
                      borderRadius={2}
                      className="loyalty-process-container sc-hwwEjo kitExE sc-frDJqD fcTlLW sc-ksYbfQ kYqsUP"
                      left="50%"
                      maxHeight="100vh"
                      maxWidth="100vw"
                      onKeyDown={[Function]}
                      p={3}
                      role="dialog"
                      top="50%"
                    >
                      <ForwardRef
                        0="s"
                        1="p"
                        2="a"
                        3="c"
                        4="e"
                        bg="white"
                        borderRadius={2}
                        className="loyalty-process-container sc-hwwEjo kitExE sc-frDJqD fcTlLW sc-ksYbfQ kYqsUP sc-TOsTZ cESxnL"
                        left="50%"
                        maxHeight="100vh"
                        maxWidth="100vw"
                        onKeyDown={[Function]}
                        p={3}
                        role="dialog"
                        top="50%"
                      >
                        <div
                          0="s"
                          1="p"
                          2="a"
                          3="c"
                          4="e"
                          className="loyalty-process-container sc-hwwEjo kitExE sc-frDJqD fcTlLW sc-ksYbfQ kYqsUP sc-TOsTZ cESxnL"
                          onKeyDown={[Function]}
                          role="dialog"
                        >
                          <div
                            className="mobile-background"
                          />
                          <Styled(Flex)
                            className="loyalty-process-account-parent"
                          >
                            <Flex
                              className="loyalty-process-account-parent sc-esjQYD cZxuvG"
                            >
                              <Box
                                className="loyalty-process-account-parent sc-esjQYD cZxuvG sc-bwzfXH hKiLMS"
                              >
                                <div
                                  className="loyalty-process-account-parent sc-esjQYD cZxuvG sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
                                >
                                  <Styled(Heading)
                                    aria-label="test"
                                    className="loyalty-process-header"
                                    id="accrual-modal-title"
                                    tabIndex={0}
                                  >
                                    <Heading
                                      aria-label="test"
                                      className="loyalty-process-header sc-gwVKww PlFLo"
                                      fontSize={5}
                                      fontWeight="bold"
                                      id="accrual-modal-title"
                                      is="h2"
                                      lineHeight={1.25}
                                      m={0}
                                      tabIndex={0}
                                    >
                                      <ForwardRef
                                        aria-label="test"
                                        className="loyalty-process-header sc-gwVKww PlFLo sc-VigVT fXkMTo"
                                        fontSize={5}
                                        fontWeight="bold"
                                        id="accrual-modal-title"
                                        is="h2"
                                        lineHeight={1.25}
                                        m={0}
                                        tabIndex={0}
                                      >
                                        <h2
                                          aria-label="test"
                                          className="loyalty-process-header sc-gwVKww PlFLo sc-VigVT fXkMTo"
                                          id="accrual-modal-title"
                                          tabIndex={0}
                                        >
                                          test
                                        </h2>
                                      </ForwardRef>
                                    </Heading>
                                  </Styled(Heading)>
                                  <Styled(Flex)
                                    className="msg-parent"
                                    success={false}
                                  >
                                    <Flex
                                      className="msg-parent sc-kIPQKe kZbVUA"
                                      success={false}
                                    >
                                      <Box
                                        className="msg-parent sc-kIPQKe kZbVUA sc-bwzfXH hKiLMS"
                                        success={false}
                                      >
                                        <div
                                          className="msg-parent sc-kIPQKe kZbVUA sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
                                        >
                                          <Styled(Label)
                                            aria-label="LOYALTY_FETCH_ERROR"
                                            className="msg-label"
                                            tabIndex={0}
                                          >
                                            <Label
                                              alignItems="center"
                                              aria-label="LOYALTY_FETCH_ERROR"
                                              className="msg-label sc-hXRMBi bAGiAn"
                                              fontSize={1}
                                              is="label"
                                              mb={1}
                                              tabIndex={0}
                                            >
                                              <ForwardRef
                                                alignItems="center"
                                                aria-label="LOYALTY_FETCH_ERROR"
                                                className="msg-label sc-hXRMBi bAGiAn sc-jKJlTe lknMeZ"
                                                fontSize={1}
                                                is="label"
                                                mb={1}
                                                tabIndex={0}
                                              >
                                                <label
                                                  aria-label="LOYALTY_FETCH_ERROR"
                                                  className="msg-label sc-hXRMBi bAGiAn sc-jKJlTe lknMeZ"
                                                  tabIndex={0}
                                                >
                                                  LOYALTY_FETCH_ERROR
                                                </label>
                                              </ForwardRef>
                                            </Label>
                                          </Styled(Label)>
                                        </div>
                                      </Box>
                                    </Flex>
                                  </Styled(Flex)>
                                  <Styled(Text)
                                    aria-label="LOYALTY_ACCRUE_ERROR"
                                    aria-live="polite"
                                    className="error-text"
                                    style={
                                      Object {
                                        "color": "red",
                                      }
                                    }
                                    tabIndex={0}
                                  >
                                    <Text
                                      aria-label="LOYALTY_ACCRUE_ERROR"
                                      aria-live="polite"
                                      className="error-text sc-iQNlJl gsloEY"
                                      m={0}
                                      style={
                                        Object {
                                          "color": "red",
                                        }
                                      }
                                      tabIndex={0}
                                    >
                                      <ForwardRef
                                        aria-label="LOYALTY_ACCRUE_ERROR"
                                        aria-live="polite"
                                        className="error-text sc-iQNlJl gsloEY sc-gqjmRU kKsieB"
                                        m={0}
                                        style={
                                          Object {
                                            "color": "red",
                                          }
                                        }
                                        tabIndex={0}
                                      >
                                        <div
                                          aria-label="LOYALTY_ACCRUE_ERROR"
                                          aria-live="polite"
                                          className="error-text sc-iQNlJl gsloEY sc-gqjmRU kKsieB"
                                          style={
                                            Object {
                                              "color": "red",
                                            }
                                          }
                                          tabIndex={0}
                                        >
                                          <div
                                            i18nKey="LOYALTY_ACCRUE_ERROR"
                                          />
                                        </div>
                                      </ForwardRef>
                                    </Text>
                                  </Styled(Text)>
                                  <Styled(Flex)>
                                    <Flex
                                      className="sc-bsbRJL fMRhsV"
                                    >
                                      <Box
                                        className="sc-bsbRJL fMRhsV sc-bwzfXH hKiLMS"
                                      >
                                        <div
                                          className="sc-bsbRJL fMRhsV sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
                                        >
                                          <Styled(Button)
                                            aria-label="RE_ENTER"
                                            className="re-enter-button"
                                            onClick={[Function]}
                                            onKeyDown={[Function]}
                                            role="button"
                                            tabIndex={0}
                                          >
                                            <Button
                                              aria-label="RE_ENTER"
                                              bg="blue"
                                              border={0}
                                              borderRadius={2}
                                              className="re-enter-button sc-iQKALj iXWBnP"
                                              color="white"
                                              fontSize={1}
                                              fontWeight="bold"
                                              is="button"
                                              lineHeight={1.1428571428571428}
                                              m={0}
                                              onClick={[Function]}
                                              onKeyDown={[Function]}
                                              px={3}
                                              py={2}
                                              role="button"
                                              tabIndex={0}
                                            >
                                              <ForwardRef
                                                aria-label="RE_ENTER"
                                                bg="blue"
                                                border={0}
                                                borderRadius={2}
                                                className="re-enter-button sc-iQKALj iXWBnP sc-ifAKCX ceNdfT"
                                                color="white"
                                                fontSize={1}
                                                fontWeight="bold"
                                                is="button"
                                                lineHeight={1.1428571428571428}
                                                m={0}
                                                onClick={[Function]}
                                                onKeyDown={[Function]}
                                                px={3}
                                                py={2}
                                                role="button"
                                                tabIndex={0}
                                              >
                                                <button
                                                  aria-label="RE_ENTER"
                                                  className="re-enter-button sc-iQKALj iXWBnP sc-ifAKCX ceNdfT"
                                                  onClick={[Function]}
                                                  onKeyDown={[Function]}
                                                  role="button"
                                                  tabIndex={0}
                                                >
                                                  <div
                                                    i18nKey="RE_ENTER"
                                                  />
                                                </button>
                                              </ForwardRef>
                                            </Button>
                                          </Styled(Button)>
                                          <Styled(Button)
                                            aria-label="PAYMENT_PAGE_FETCHING_CLOSE"
                                            className="close-button"
                                            onClick={[Function]}
                                            role="button"
                                            tabIndex={0}
                                          >
                                            <Button
                                              aria-label="PAYMENT_PAGE_FETCHING_CLOSE"
                                              bg="blue"
                                              border={0}
                                              borderRadius={2}
                                              className="close-button sc-iQKALj iXWBnP"
                                              color="white"
                                              fontSize={1}
                                              fontWeight="bold"
                                              is="button"
                                              lineHeight={1.1428571428571428}
                                              m={0}
                                              onClick={[Function]}
                                              px={3}
                                              py={2}
                                              role="button"
                                              tabIndex={0}
                                            >
                                              <ForwardRef
                                                aria-label="PAYMENT_PAGE_FETCHING_CLOSE"
                                                bg="blue"
                                                border={0}
                                                borderRadius={2}
                                                className="close-button sc-iQKALj iXWBnP sc-ifAKCX ceNdfT"
                                                color="white"
                                                fontSize={1}
                                                fontWeight="bold"
                                                is="button"
                                                lineHeight={1.1428571428571428}
                                                m={0}
                                                onClick={[Function]}
                                                px={3}
                                                py={2}
                                                role="button"
                                                tabIndex={0}
                                              >
                                                <button
                                                  aria-label="PAYMENT_PAGE_FETCHING_CLOSE"
                                                  className="close-button sc-iQKALj iXWBnP sc-ifAKCX ceNdfT"
                                                  onClick={[Function]}
                                                  role="button"
                                                  tabIndex={0}
                                                >
                                                  <div
                                                    i18nKey="PAYMENT_PAGE_FETCHING_CLOSE"
                                                  />
                                                </button>
                                              </ForwardRef>
                                            </Button>
                                          </Styled(Button)>
                                        </div>
                                      </Box>
                                    </Flex>
                                  </Styled(Flex)>
                                </div>
                              </Box>
                            </Flex>
                          </Styled(Flex)>
                          <Styled(Component)
                            aria-label="EXIT_DIALOG"
                            className="close-btn"
                            onClick={[Function]}
                            role="button"
                            tabIndex={0}
                          >
                            <Component
                              aria-label="EXIT_DIALOG"
                              className="close-btn sc-epnACN bZbihZ"
                              onClick={[Function]}
                              role="button"
                              tabIndex={0}
                            >
                              <IconButton
                                aria-label="EXIT_DIALOG"
                                className="close-btn sc-epnACN bZbihZ"
                                onClick={[Function]}
                                role="button"
                                tabIndex={0}
                              >
                                <Styled(ButtonCircle)
                                  aria-label="EXIT_DIALOG"
                                  className="close-btn sc-epnACN bZbihZ"
                                  onClick={[Function]}
                                  role="button"
                                  tabIndex={0}
                                >
                                  <ButtonCircle
                                    aria-label="EXIT_DIALOG"
                                    bg="blue"
                                    border={0}
                                    borderRadius={9999}
                                    className="close-btn sc-epnACN bZbihZ sc-tilXH jponlC"
                                    color="white"
                                    extend={[Function]}
                                    fontSize={1}
                                    fontWeight="bold"
                                    is="button"
                                    lineHeight={1.1428571428571428}
                                    m={0}
                                    onClick={[Function]}
                                    px={3}
                                    py={2}
                                    role="button"
                                    tabIndex={0}
                                  >
                                    <ForwardRef
                                      aria-label="EXIT_DIALOG"
                                      bg="blue"
                                      border={0}
                                      borderRadius={9999}
                                      className="close-btn sc-epnACN bZbihZ sc-tilXH jponlC sc-bZQynM bOrPFB"
                                      color="white"
                                      extend={[Function]}
                                      fontSize={1}
                                      fontWeight="bold"
                                      is="button"
                                      lineHeight={1.1428571428571428}
                                      m={0}
                                      onClick={[Function]}
                                      px={3}
                                      py={2}
                                      role="button"
                                      tabIndex={0}
                                    >
                                      <Button
                                        aria-label="EXIT_DIALOG"
                                        bg="blue"
                                        border={0}
                                        borderRadius={9999}
                                        className="close-btn sc-epnACN bZbihZ sc-tilXH jponlC sc-bZQynM bOrPFB"
                                        color="white"
                                        fontSize={1}
                                        fontWeight="bold"
                                        is="button"
                                        lineHeight={1.1428571428571428}
                                        m={0}
                                        onClick={[Function]}
                                        px={3}
                                        py={2}
                                        role="button"
                                        tabIndex={0}
                                      >
                                        <ForwardRef
                                          aria-label="EXIT_DIALOG"
                                          bg="blue"
                                          border={0}
                                          borderRadius={9999}
                                          className="close-btn sc-epnACN bZbihZ sc-tilXH jponlC sc-bZQynM bOrPFB sc-ifAKCX etSojC"
                                          color="white"
                                          fontSize={1}
                                          fontWeight="bold"
                                          is="button"
                                          lineHeight={1.1428571428571428}
                                          m={0}
                                          onClick={[Function]}
                                          px={3}
                                          py={2}
                                          role="button"
                                          tabIndex={0}
                                        >
                                          <button
                                            aria-label="EXIT_DIALOG"
                                            className="close-btn sc-epnACN bZbihZ sc-tilXH jponlC sc-bZQynM bOrPFB sc-ifAKCX etSojC"
                                            onClick={[Function]}
                                            role="button"
                                            tabIndex={0}
                                          >
                                            <i />
                                            ✕
                                          </button>
                                        </ForwardRef>
                                      </Button>
                                    </ForwardRef>
                                  </ButtonCircle>
                                </Styled(ButtonCircle)>
                              </IconButton>
                            </Component>
                          </Styled(Component)>
                        </div>
                      </ForwardRef>
                    </Position>
                  </ForwardRef>
                </Fixed>
              </ForwardRef>
            </Modal>
          </Styled(Modal)>
        </div>
      </Box>
    </Flex>
  </LoyaltyProcess>
</ThemeProvider>
`;
